### ISSUE TITLE
Attribute "content-type" on p:inline/p:document after static variables

### ISSUE THREAD

**Issue by xml-project** [xproc/3.0-specification#470]
If I remember right, we introduced @content-type on p:inline/p:document so we can determine the document type statically, while @document-properties is determined dynamically.
Now we have static variables/options and @document-properties on p:inline/p:document is evaluated during static analysis:
Do we still need @content-type or could we rely just on the entry in @document-properties?

**Comment by ndw** [comment]
Hmm. I'm confused again. We really seem to have tangled ourselves up on this issue. I prefer @content-type because it's easier to type and I think it'll be fairly common. I think we have rules now for static analysis of inputs/outputs/etc. on p:declare-step. It's not immediately obvious to me that this needs to apply globally. Consider:

```xml
<p:declare-step version="3.0" xmlns:p="http://www.w3.org/ns/xproc" name="main">
  <p:option name="method" static="true" select="'xml'"/>
  <p:output port="result" serialization="map { 'method': $method }"/>
  <p:option name="href"/>
  <p:option name="ctype"/>

  <p:identity>
    <p:input port="source">
      <p:document href="{$href}" content-type="{$ctype}"/>
    </p:input>
  </p:identity>
</p:declare-step>
```

Why did we decide to forbid that?

**Comment by xml-project** [comment]
Yes, I am confused to. In the current specs we [say](http://spec.xproc.org/master/head/xproc/#static-expressions)

> Several kinds of expressions are evaluated during static analysis: [...]
>3. The document-properties of p:inline and p:document elements.

and `@content-type` is no AVT on p:inline and p:document. 

So the content-type can only be set statically. `content-type="{$ctype}"` (from the example) will be an error, because it is not a valid form of a content type.

Beside: Why is document-properties to be determined static? Intuitively would prefer to keep it dynamic, so I can use it to pass (non-static) option values to steps using the map. But as we decided it to be static, there is obviously something I have overlooked something, but what?

The reason to make content-type static what issue #426, but that was specific for p:inline, so may be we got it wrong for p:document. Keeping in mind, that we define p:document via p:load, reading the specs let me concluded, that the content-type for p:document can only be set statically (see quoted rules), but it can not be determined statically. In p:load we say, that if there is neither a @content-type nor an respective entry in @document-properties, that the content-type is determined by the content-type delivered from the file-system (http: etc.)

Apart from the static/dynamic problem, I would agree to keep `content-type` as syntactic sugar, because it is easier to write `content-type="text/plain"` then `document-properties="map{'content-type':'text/plain'}"` -- Hope I understood @ndw 's argument right.
